src/stored-cache.ts

```diff
@@ -0,0 +1,56 @@
+import { match } from "ts-pattern";
+import {
+  memoryCache,
+  CacheOptions as MemoryCacheOptions,
+} from "./memory-cache";
+import { CacheUpdatePolicy } from "./policies";
+
+export type JSONLiteral = String | Number | Boolean | null;
+export type JSONObject = { [key: string]: Json };
+export type Json = JSONLiteral | JSONObject | Json[];
+
+export interface IStorage<TKey, TData extends Json> {
+  getItem: (
+    key: TKey
+  ) => Promise<{ value: TData; metadata: { lastUpdatedAt: Date } } | null>;
+  setItem: (
+    key: TKey,
+    value: TData,
+    metadata: { lastUpdatedAt: Date }
+  ) => Promise<void>;
+}
+
+export type CacheOptions<TKey, TData extends Json> = MemoryCacheOptions<
+  TData,
+  TKey[]
+> & {
+  storage: IStorage<TKey, TData>;
```

Comment:
Why is it cool it intersect options with storage instead of having a storage parameter in `storedCache`?

Reply:
@kasperhj You mean as a seperate argument/parameter rather an options key? 
I generally prefer just extending options, but no strong feelings